src/nvim/drawscreen.c

```diff
@@ -2681,6 +2681,11 @@ void status_redraw_buf(buf_T *buf)
       redraw_later(wp, UPD_VALID);
     }
   }
+  // Redraw the ruler if it is in the command line and was not marked for redraw above
+  if (p_ru && one_nonfloat() && !curwin->w_status_height && !curwin->w_redr_status) {
```

Comment:
I'm not sure these conditions are too exclusive or not exclusive enough for what's necessary but the issue here is that `redrawstatus` is responsible for redrawing the ruler but `status_redraw_buf()` only checks for the statusline.

It also seems to me that the above statusline conditions should be kept in sync with `status_redraw_all()` and `redraw_buf_status_later()`. They are not identical currently.

Reply:
at least, `one_nonfloat()` should be removed from here. with laststatus=0 this could happen for a lower split window. `!curwin->w_status_height` is enough to check for this condition.